pkg/asset/internal/templates.go

```diff
@@ -141,6 +141,7 @@ spec:
         - --tls-private-key-file=/etc/kubernetes/secrets/apiserver.key
         - --service-account-key-file=/etc/kubernetes/secrets/service-account.pub
         - --client-ca-file=/etc/kubernetes/secrets/ca.crt
+        - --cloud-provider={{ .CloudProvider }}
```

Comment:
Oh, I have nearly the same PR for our fork (mistakenly thought you weren't doing this) and I have the check